lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -359,7 +359,12 @@ class AwsCompileFunctions {
         [versionLogicalId]: newVersion,
       };
 
-      if (this.serverless.service.provider.versionFunctions) {
+      const versionFunction = functionObject.versionFunction === undefined ||
+        functionObject.versionFunction === null
+          ? this.serverless.service.provider.versionFunctions
+          : functionObject.versionFunction;
```

Comment:
I think this change is wrong and changes the existing behavior of `provider.versionFunctions`. It now will default to true, if the flas hasn't been set on the function, regardless how the service wide flag is set.

(1) If someone sets `provider.versionFunctions: false` it must be turned off, if no flag is given on a function - provider.versionFunctions must be respected as default.
(2) The old default was have it turned off. This would change the default and thus be a breaking change.

We should make sure that this is not breaking and that the hierarchy of definitions is respected properly.

Reply:
Oh, it is possible that I've made a mistake there. I'll check that. The idea is that in `versionFunction` in function definition overwrites the `versionFunctions` if it is set, like other variables (environment etc.), otherwise, it should work as before. So it is not a breaking change. But I'll check the code. The lambda@edge functions need to be versioned, that is why I'm changing this behaviour.

Reply:
Yeah, that is also my expectation of the behavior.